File: 프로그래머스/2/131127. 할인 행사/할인 행사.py
from collections import Counter

def solution(want, number, discount):
    answer = 0
    want_dict = {w: n for w, n in zip(want, number)}
    for i in range(len(discount) - 9): 
        current_discount_slice = discount[i : i + 10]
        current_discount_counts = Counter(current_discount_slice)
        can_buy_all = True
        for item, required_count in want_dict.items():
            if current_discount_counts.get(item, 0) < required_count:
                can_buy_all = False
                break
        if can_buy_all:
            answer += 1        
    return answer

File: 프로그래머스/2/131127. 할인 행사/할인 행사.js
function solution(want, number, discount) {
    let numbers = [];
    let answer = 0, count = 0;
    
    for (let i = 0; i <= discount.length - 10; i++) {
        numbers = [...number];
        const stuff = discount.slice(i, i + 10);
        for (const el of stuff) {
            const idx = want.indexOf(el);
            if (idx === -1) continue;
            if (numbers[idx] > 0) {
                numbers[idx]--;
                count++;
            }
        }
        if (count === 10) answer++;
        count = 0;
    }
    return answer;
}